lib/cc/analyzer/formatters/plain_text_formatter.rb

```diff
@@ -42,11 +42,11 @@ def engine_running(engine, &block)
           end
         end
 
-        def failed(output)
-          spinner.stop("Failed")
-          puts colorize("\nAnalysis failed with the following output:", :red)
+        def errored(output)
+          spinner.stop("Errored")
+          puts colorize("\nAnalysis errored with the following output:", :red)
           puts output
-          exit 1
+          exit 2
```

Comment:
I like it, but I'm not sure how much mileage we'd get out of that idea since most of the individual error codes can be determined externally. A non-zero exit from analyze and validate-config are probably the only two cases that could use differentiation.